packages/react/src/components/Modal/Modal.base.tsx

```diff
@@ -228,6 +229,18 @@ export const ModalBase: React.FunctionComponent<IModalProps> = React.forwardRef<
       [keepInBounds, internalState],
     );
 
+    /**
+     * Determines the modal role for accessibility
+     * The isModeless and isBlocking will be ignored only when isAlert is set
+     */
+    const getDialogRole = () => {
+      if (isAlert || isModeless === false || isBlocking) {
```

Comment:
In addition to @khmakoto's comment, we also want to let people specify `isAlert="false"` to opt out of the `alertdialog` role, even for blocking modal dialogs.

I'd suggest giving `isAlert` a default value of true/false based on `isBlocking && !isModeless`, then instead of `getDialogRole`, it would just be `role={isAlert ? 'alertdialog' : 'dialog'}`

Reply:
Yes definitely, added this ```const isAlertRole = isAlert ?? (isBlocking && !isModeless);``` to set default value from props only if the ```isAlert``` is undefined